app/controllers/species_list_controller.rb

```diff
@@ -995,44 +926,60 @@ def find_chosen_name(id, alternatives)
   # an Array of names where the values are [display_name, name_id].  This
   # is destined for the instance variable @checklist.
   def calc_checklist(query = nil)
-    results = []
-    if query || (query = query_from_session)
-      results = case query.model
-                when Name
-                  query.select_rows(
-                    select: "DISTINCT names.display_name, names.id",
-                    limit: 1000
-                  )
-                when Observation
-                  query.select_rows(
-                    select: "DISTINCT names.display_name, names.id",
-                    join: :names,
-                    limit: 1000
-                  )
-                when Image
-                  query.select_rows(
-                    select: "DISTINCT names.display_name, names.id",
-                    join: { images_observations: { observations: :names } },
-                    limit: 1000
-                  )
-                when Location
-                  query.select_rows(
-                    select: "DISTINCT names.display_name, names.id",
-                    join: { observations: :names },
-                    limit: 1000
-                  )
-                when RssLog
-                  query.select_rows(
-                    select: "DISTINCT names.display_name, names.id",
-                    join: { observations: :names },
-                    where: "rss_logs.observation_id > 0",
-                    limit: 1000
-                  )
-                else
-                  []
-                end
+    return unless query ||= query_from_session
+
+    case query.model.name
```

Comment:
Only the Observation case is covered.  Given the refactor it would be good to get these other cases covered since code varies from case to case.